App.Cards/Repositories/EFCardsRepository.cs

```diff
@@ -0,0 +1,112 @@
+﻿using App.Cards.Database;
+using App.Cards.Exceptions;
+using App.Configuration;
+using App.Models.Cards;
+using App.Repositories;
+using System;
+using System.Linq;
+
+namespace App.Cards.Repositories
+{
+    public class EFCardsRepository: ICardsRepository, IDisposable, ITransientDependency
+    {
+        private readonly CardsDbContext _dbContext;
+
+        public EFCardsRepository(CardsDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        private Card GetCardByNumber(long number)
+        {
+            var cards = _dbContext.Cards.ToList();
+
+            Card result = null;
+
+            foreach (var card in cards)
+            {
+                if (card.Number == number)
+                {
+                    result = card;
+                    break;
+                }
+            }
+
+            return result;
+        }
+
+        private Card CheckCardByCVV(Card card, DateTime expiresEnd, ushort CVV)
+        {
+            if (card != null)
+            {
+                if (card.ExpiresEnd != expiresEnd || card.CVV != CVV)
+                    card = null;
+            }
+
+            return card;
+        }
+
+        
+
+        public Card GetCard(long number, DateTime expiresEnd, ushort CVV)
+        {
+            Card result = GetCardByNumber(number);
+
+            result = CheckCardByCVV(result, expiresEnd, CVV);
```

Comment:
it is better to first check by null and then perform any operation

Reply:
check by null occurs in the method GetCard